api/src/neo4j/queries/externalDb.js

```diff
@@ -1,10 +1,22 @@
 import querySingleResult from 'neo4j/queryHandlers/single';
 
 const getComponentsForExternalDb = async ({ dbName, externalId }) => {
-  const statement = `
+  let statement = `
 MATCH (db:ExternalDb {dbName: '${dbName}', externalId: '${externalId}'})-[v]-(c)
 RETURN { externalDb: properties(db), components: COLLECT({ component: c, version: type(v) }) }
 `;
+  if (dbName === 'MetabolicAtlas') {
+    statement = `
+MATCH (r:Reaction {id: '${externalId}'})-[v]-(c)
+RETURN { externalDb: properties(r), components: COLLECT(DISTINCT({component: r, version: type(v)}))}
+
+UNION
+
+MATCH (r:CompartmentalizedMetabolite {id: '${externalId}'})-[v]-(c)
```

Comment:
or here:

```suggestion
MATCH (r:CompartmentalizedMetabolite {id: '${externalId}'})-[v]-()
```

Reply:
Minor detail, but I would rename `r` to `cm`

Reply:
It seems one can not use different column names when `UNION` is used @inghylt. I got this error ```All sub queries in an UNION must have the same column names (line 5, column 1 (offset: 147)) "UNION" ^``` if I change `r` to `cm`

Reply:
https://neo4j.com/docs/cypher-manual/current/clauses/union/

Yup, that seems to be the case! Good to know